File: Projekti/C#-Native-Benchmark/Repository/PostRepository.cs
using MySql.Data.MySqlClient;
using Native.Repository.Interface;
using Native.Settings;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Native.Repository
{
    public class PostRepository : IPostRepository
    {
        private readonly string _connectionString;

        public PostRepository(IDatabaseConnection connection)
        {
            _connectionString = connection.ConnectionString;
        }

        public void UpdateUserPosts(int userId, string text)
        {
            MySqlConnection connection = new(_connectionString);
            connection.Open();
            MySqlCommand command = connection.CreateCommand();

            command.CommandText = "UPDATE Posts SET content = @text WHERE user_id = @id";
            command.Parameters.AddWithValue("@id", userId);
            command.Parameters.AddWithValue("@text", text);

            command.ExecuteNonQuery();
        }
    }
}

File: Projekti/Python-Native-Benchmark/repository/post_repository.py
class PostRepository:
    def __init__(self, conn):
        self.conn = conn

    def update_posts_for_user(self, user_id, content):

        query = "UPDATE Posts SET content = %s WHERE user_id = %s"
        

        with self.conn.cursor() as cursor: 
            cursor.execute(query, (content, user_id,))

        self.conn.commit()